### PR TITLE
feat(seid): ConfigManager selection seam (PLT-775 PR1)

### PR DESCRIPTION
**Pull request by bdchatham**
## What

PR1 of the ConfigManager MVP (**PLT-775**): the **logical seam** between the legacy config loader and the forthcoming sei-config-backed manager, gated by `SEI_CONFIG_MANAGER`. This PR ships the seam only — no behavior change on the default path.

New package `cmd/seid/cmd/configmanager`:
- `ConfigManager` interface — `Apply(cmd, customAppConfigTemplate, customAppConfig)` mirrors `server.InterceptConfigsPreRunHandler`.
- `LegacyConfigManager` — re-enters the legacy handler **verbatim** (the default).
- `SeiConfigManager` — PR1 **stub**: returns a hard error `SEI_CONFIG_MANAGER=v2 not yet implemented (PR1 seam only)`; **does not import sei-config**.
- `Select(getenv)` — `unset`/`legacy` → Legacy; `v2` → Sei; **any other value → hard error** (no silent fallback).

`root.go` `PersistentPreRunE` now routes through `Select`+`Apply` instead of calling `InterceptConfigsPreRunHandler` directly. The `init` skip is untouched.

## Scope / constraints

- **Zero edits to the vendored `sei-cosmos` fork.** Diff is confined to `cmd/seid/cmd/` (the new package + a 6-line wiring change in `root.go`).
- One-day seam slice; the v2 body (sei-config wiring) is a follow-up PR.

## Acceptance-criteria drift contract

Per the canonical design (the design doc [`designs/config-manager/DESIGN.md`](https://github.com/sei-protocol/bdchatham-designs/blob/main/designs/config-manager/DESIGN.md) — *The selection seam* / *MVP scope*), this PR:

**Satisfies:**
- **Legacy untouched** (structural) — when `SEI_CONFIG_MANAGER` is unset, dispatch routes to the unmodified `InterceptConfigsPreRunHandler`.
- **No silent fallback** — any value other than unset/`legacy`/`v2` is a hard startup error.

**Explicitly deferred to PR2+** (all require sei-config as a live dependency):
- v2 parity (legacy-vs-v2 differential on parsed semantics)
- fresh/mixed-home matrix
- `SEI_` collision audit
- `sc-write-mode` failure-parity (the PLT-775 incident class, extending into `app.New()`)
- `KNOWN_UNMAPPED_FIELDS` enumeration

## Tests

`cmd/seid/cmd/configmanager` table test: dispatch (unset / legacy / v2 / garbage→error) + the v2-stub-errors assertion. `go build ./cmd/seid/...` green.

## Lineage

- Issue: **PLT-775** (rides the issue directly; this is the first slice of the single MVP deliverable).
- Design: [`designs/config-manager/DESIGN.md`](https://github.com/sei-protocol/bdchatham-designs/blob/main/designs/config-manager/DESIGN.md) (bdchatham-designs).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmd/seid/cmd/configmanager/configmanager_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	cases := []struct {
+		name    string
+		val     string
+		want    ConfigManager
+		wantErr bool
+	}{
+		{name: "unset", val: "", want: LegacyConfigManager{}},
+		{name: "legacy", val: "legacy", want: LegacyConfigManager{}},
+		{name: "v2", val: "v2", want: SeiConfigManager{}},
+		{name: "garbage", val: "v3", wantErr: true},
```
Comment (amir-deris) [inline cmd/seid/cmd/configmanager/configmanager_test.go:22]: Perhaps can include test cases for empty string?

**Review by amir-deris — APPROVED** [review]
